src/aggregation/intermediate_agg_result.rs

```diff
@@ -246,22 +255,75 @@ impl IntermediateTermBucketResult {
         self,
         req: &TermsAggregation,
         sub_aggregation_req: &AggregationsInternal,
-    ) -> BucketResult {
+    ) -> crate::Result<BucketResult> {
         let req = TermsAggregationInternal::from_req(req);
         let mut buckets: Vec<BucketEntry> = self
             .entries
             .into_iter()
             .filter(|bucket| bucket.1.doc_count >= req.min_doc_count)
-            .map(|(key, entry)| BucketEntry {
-                key: Key::Str(key),
-                doc_count: entry.doc_count,
-                sub_aggregation: AggregationResults::from_intermediate_and_req_internal(
-                    entry.sub_aggregation,
-                    sub_aggregation_req,
-                ),
+            .map(|(key, entry)| {
+                Ok(BucketEntry {
+                    key: Key::Str(key),
+                    doc_count: entry.doc_count,
+                    sub_aggregation: AggregationResults::from_intermediate_and_req_internal(
+                        entry.sub_aggregation,
+                        sub_aggregation_req,
+                    )?,
+                })
             })
-            .collect();
-        buckets.sort_unstable_by_key(|bucket| std::cmp::Reverse(bucket.doc_count));
+            .collect::<crate::Result<_>>()?;
+
+        let order = req.order.order;
+        match req.order.target {
+            OrderTarget::Key => {
+                buckets.sort_by(|bucket1, bucket2| {
+                    if req.order.order == Order::Desc {
+                        bucket1
```

Comment:
very local, but there is probably a nicer way to write this.

Reply:
I did a minor improvement on it